acceptance/Rakefile

```diff
@@ -0,0 +1,58 @@
+def beaker(command)
+  debug_flag = ''
+  if ENV['BEAKER_debug'] || ENV['DEBUG']
+    debug_flag = '--debug'
+  end
+
+  sh(%(beaker #{command} #{debug_flag}))
+end
+
+desc 'Prepare for running tests: Run beaker init, provision, and pre-suite all at once.'
+task :prepare do
+  errors = []
+
+  if !ENV['MASTER_PACKAGE_VERSION'] && !ENV['MASTER_COLLECTION']
+    errors << 'You must set a starting version of puppet-agent using ' +
+              '$MASTER_PACKAGE_VERSION or $MASTER_COLLECTION'
+  end
+
+  if ENV['HOSTS']
+    hosts_file = File.expand_path(ENV['HOSTS'])
+    errors << "Couldn't find $HOSTS file at #{hosts_file}" unless File.exist?(hosts_file)
+  elsif File.exist?('hosts.yml')
+    hosts_file = './hosts.yml'
+  elsif File.exist?('hosts.yaml')
+    hosts_file = './hosts.yaml'
+  else
+    errors << 'Unable to find a hosts file in $HOSTS or at ./hosts.yml or at ./hosts.yaml'
+  end
+
+  unless errors.empty?
+    raise errors.join("\n")
+  end
+
+  puts "Using beaker hosts file #{hosts_file}"
+
+  # Log the contents of the host file if running in Jenkins
+  if ENV['JENKINS_HOME']
+    pp File.read(hosts_file)
+  end
+
+  beaker("init -h #{hosts_file} -o options.rb")
+  beaker("provision")
+  beaker("exec ./pre_suite")
```

Comment:
I found during my work adding only-fails to beaker(-puppet) that if one of the beaker subcommands fails, it will kill the rake run right then and there, so no subsequent beaker subcommands (or indeed anything else in the Rakefile in question) will be executed. This means that provisioned machines will wait for their timeout before they are reaped.

I addressed this by wrapping things with a 'begin/rescue/ensure' block, as illustrated here: https://github.com/puppetlabs/beaker-puppet/blob/1d269589adbb9c47888e47797e63c749133a58a7/tasks/ci.rake#L275-L297

Reply:
I'm not sure I would want that to happen as a _developer_ using these tests -- I would want to look at the SUTs and see what happened and then manually destroy them -- but I'll add that to the CI task now.

Reply:
I see your point. Would we want to add the complexity of making that conditional, along the line of the -`-preserve-hosts` flag?

Reply:
I agree that an option to destroy the hosts would be useful.

Reply:
This is one of the many places where the beaker-with-subcommands vs. beaker-without-subcommands workflows diverge. `--preserve-hosts` does not make a lot of sense with subcommands unless you're running the entire test suite with one invocation, since `beaker destroy` is already available and intended to be called separately. I'm hoping this is something we can refine in subsequent PRs.